src/NuGet.Core/NuGet.Commands/RestoreCommand/ProjectRestoreCommand.cs

```diff
@@ -324,7 +325,7 @@ private async Task<bool> ResolutionSucceeded(IEnumerable<RestoreTargetGraph> gra
             CancellationToken token)
         {
             var packagesToInstall = graphs
-                .SelectMany(g => g.Install.Where(match => uniquePackages.Add(match.Library))).ToList();
+                .SelectMany(g => g.Install.Where(match => uniquePackages.Add(match.Library))).AsList();
```

Comment:
This change is unnecessary. There's no functional change here. 
LINQ generates enumerates which had to be converted to List.